### PR TITLE
Add controller action for posting browser metric events

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Extends the existing browser events controller to add functionality to post metrics (to the [PutMetricData](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricData.html) Cloudwatch endpoint).

Example metrics in Cloudwatch:

Screenshot 2023-06-01 at 3 59 06 PM

## Links

https://codedotorg.atlassian.net/browse/SL-800

## Testing story

Tested locally with front end changes (future PR). Verified metrics appear correctly.

## Follow-up work

Front end implementation: https://codedotorg.atlassian.net/browse/SL-801

### CODE REVIEW

**Review by bencodeorg — APPROVED** [review]

File: dashboard/app/controllers/browser_events_controller.rb
Comment (jmkulwik) [inline dashboard/app/controllers/browser_events_controller.rb:68]: Why?

Comment (sanchitmalhotra126) [reply dashboard/app/controllers/browser_events_controller.rb:68]: The [AWS Ruby SDK](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/CloudWatch/Client.html#put_metric_data-instance_method) function looks for this specific parameter. I'll add a comment to clarify.

**Comment by pablo-code-org** [comment]
Can you check Cdo::metrics  and also Infrastructure::Logger

They are doing similar things. Cdo will be more useful, but ideally we should have one solution

**Comment by sanchitmalhotra126** [comment]
> Can you check Cdo::metrics and also Infrastructure::Logger
> 
> They are doing similar things. Cdo will be more useful, but ideally we should have one solution

Oh good call, Cdo::Metrics looks like it'll work well. thanks!

**Comment by pablo-code-org** [comment]
I used Cdo::metrics and created the wrapper around it. The benefit I found was that it handles buffering the events.

The downside is that it is not async (which not a big deal so far, but wondering if an async flush would be nicer)

File: dashboard/app/controllers/browser_events_controller.rb
Comment (pablo-code-org) [inline dashboard/app/controllers/browser_events_controller.rb:54]: where do you flush the data? Not sure if it will to the server if you dont call flush

Comment (sanchitmalhotra126) [reply dashboard/app/controllers/browser_events_controller.rb:54]: Oh do we need to flush the buffer manually? I thought it would automatically flush after its batch size had been reached? I tried manually reducing the batch size to 2 and I thought I saw metrics post but maybe I'm misunderstanding.

Comment (sanchitmalhotra126) [reply dashboard/app/controllers/browser_events_controller.rb:54]: Resolved this on Slack

**Review by pablo-code-org — COMMENTED** [review]
Approving after merging :) I just fixed my github account logging :')

**Pull request merged**
